src/pages/achievement/control/subcomponents/editorTools/EditableAchievementCard.tsx

```diff
@@ -0,0 +1,221 @@
+import { Card } from '@blueprintjs/core';
+import React, { useState } from 'react';
+
+import {
+  AchievementAbility,
+  AchievementGoal,
+  AchievementItem,
+  AchievementViewItem
+} from '../../../../../features/achievement/AchievementTypes';
+import AchievementInferencer from '../../../dashboard/subcomponents/utils/AchievementInferencer';
+import AchievementDeleter from './editableUtils/AchievementDeleter';
+import AchievementUploader from './editableUtils/AchievementUploader';
+import EditableAchievementAbility from './editableUtils/EditableAchievementAbility';
+import EditableAchievementBackground from './editableUtils/EditableAchievementBackground';
+import EditableAchievementDate from './editableUtils/EditableAchievementDate';
+import EditableAchievementTitle from './editableUtils/EditableAchievementTitle';
+import EditableAchievementGoals from './editableUtils/goals/EditableAchievementGoals';
+import EditableAchievementView from './editableView/EditableAchievementView';
+
+type EditableAchievementCardProps = {
+  achievement: AchievementItem;
+  inferencer: AchievementInferencer;
+  updateAchievements: any;
+  editAchievement: any;
+  forceRender: any;
+  adderId: number;
+  setAdderId: any;
+  addUnsavedChange: any;
+  removeUnsavedChange: any;
+  removeGoal: any;
+  removeAchievement: any;
+};
+
+function EditableAchievementCard(props: EditableAchievementCardProps) {
+  const {
+    achievement,
+    inferencer,
+    updateAchievements,
+    editAchievement,
+    adderId,
+    setAdderId,
+    addUnsavedChange,
+    removeUnsavedChange,
+    removeGoal,
+    removeAchievement
+  } = props;
+
+  const [editableAchievement, setEditableAchievement] = useState<AchievementItem>(achievement);
+  const { id, title, ability, deadline, cardTileUrl, release, goals } = editableAchievement;
+
+  const [hasChanges, setHasChanges] = useState<boolean>(false);
+  const [pendingUpload, setPendingUpload] = useState<boolean>(false);
+
+  const setUnsaved = () => {
+    if (!hasChanges) {
+      setHasChanges(true);
+      addUnsavedChange();
+    }
+  };
+
+  const setUpload = () => {
+    setPendingUpload(false);
+    removeUnsavedChange();
+  };
+
+  const handleSaveChanges = () => {
+    inferencer.modifyAchievement(editableAchievement);
+    setHasChanges(false);
+    setPendingUpload(true);
+
+    if (id === adderId) {
```

Comment:
Why is this the case? Please comment on it. Thanks!

Reply:
Commented!